### ISSUE TITLE
Update jp2 files

### ISSUE THREAD

**Issue by alunacob** [opengeospatial/ets-gmljpx20#29]
I believe the files in the repo, e.g.: src/test/resources/jp2/romagmljp2-collection2-rreq7.jp2

do not pass the tests. Shall they be updated?

**Comment by dstenger** [comment]
As the tests using this file are passing, it is not necessary to update the file.

However, unit tests might be enhanced by also using a file passing all tests.

**Comment by EmDevys** [comment]
I guess Lucio has provided a sample such as the attached file (gmljp2, zipped with 7z) which should validate.
Unfortunately the examples under http://schemas.opengis.net/gmljp2/2.0/examples/ are invalid, because they have an empty rangeType element…
I kindly requested these examples to be corrected.
Hoping this is now clear.
[roma4326-2b.zip](https://github.com/opengeospatial/ets-gmljpx20/files/3837731/roma4326-2b.zip)

**Comment by dstenger** [comment]
@bermud Can you please advise how to exchange examples at schema.opengis.net?

@EmDevys Thank you for the clarification. Also, I do not oppose to update the test resources of the test suite if the unit tests do not break then.

@keshav-nangare Can you please exchange the test resources and check if all unit tests are still passing?

**Comment by keshavnangare** [comment]
@dstenger 

I have replaced the test resource and following tests are getting failed:

from class org.opengis.cite.gmljpx20.core.VerifyCoreTests.

     - rootInstancecontainsGmlRangeTypeDataRecordUom
     - rootInstancecontainsUomByReference

**Comment by dstenger** [comment]
@keshav-nangare Can you please add the complete failure message to this issue?

**Comment by keshavnangare** [comment]
Stacktrace of failure:
```
java
org.opengis.cite.gmljpx20.core.VerifyCoreTests
rootInstancecontainsGmlRangeTypeDataRecordUom(org.opengis.cite.gmljpx20.core.VerifyCoreTests)
java.lang.AssertionError: Expected exception: java.lang.AssertionError

	at org.junit.internal.runners.statements.ExpectException.evaluate(ExpectException.java:32)

	at org.junit.rules.ExpectedException$ExpectedExceptionStatement.evaluate(ExpectedException.java:239)

	at org.junit.rules.RunRules.evaluate(RunRules.java:20)

	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)

	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)

	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)

	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)

	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)

	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)

	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)

	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)

	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)

	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)

	at org.eclipse.jdt.internal.junit4.runner.JUnit4TestReference.run(JUnit4TestReference.java:89)

	at org.eclipse.jdt.internal.junit.runner.TestExecution.run(TestExecution.java:41)

	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:541)

	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:763)

	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.run(RemoteTestRunner.java:463)

	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.main(RemoteTestRunner.java:209)



rootInstancecontainsUomByReference(org.opengis.cite.gmljpx20.core.VerifyCoreTests)
java.lang.NullPointerException

	at org.opengis.cite.gmljpx20.core.CoreTests.containsUomByReference(CoreTests.java:440)

	at org.opengis.cite.gmljpx20.core.VerifyCoreTests.rootInstancecontainsUomByReference(VerifyCoreTests.java:112)

	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)

	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)

	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)

	at java.lang.reflect.Method.invoke(Unknown Source)

	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)

	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)

	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)

	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)

	at org.junit.rules.ExpectedException$ExpectedExceptionStatement.evaluate(ExpectedException.java:239)

	at org.junit.rules.RunRules.evaluate(RunRules.java:20)

	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)

	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)

	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)

	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)

	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)

	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)

	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)

	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)

	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)

	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)

	at org.eclipse.jdt.internal.junit4.runner.JUnit4TestReference.run(JUnit4TestReference.java:89)

	at org.eclipse.jdt.internal.junit.runner.TestExecution.run(TestExecution.java:41)

	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:541)

	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:763)

	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.run(RemoteTestRunner.java:463)

	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.main(RemoteTestRunner.java:209)




```

**Comment by dstenger** [comment]
Can you try to find out why the tests are failing with new data but not with the old one?
Please document your results here.

**Comment by keshavnangare** [comment]
> rootInstancecontainsGmlRangeTypeDataRecordUom(org.opengis.cite.gmljpx20.core.VerifyCoreTests)
java.lang.AssertionError: Expected exception: java.lang.AssertionError

The test is expecting the assertion error as in older jp2 file there is no `uom` element is not defined so it will throw the exception if element not found.
And in new jp2 file has the uom element so its not throwing any exception so test gets failed as its expects the assertion error.

We can see the following snippet:
https://github.com/opengeospatial/ets-gmljpx20/blob/d861bf3c930800f318db0ea2be79c2af5c1d4565/src/test/java/org/opengis/cite/gmljpx20/core/VerifyCoreTests.java#L94-L103

> rootInstancecontainsUomByReference(org.opengis.cite.gmljpx20.core.VerifyCoreTests)
java.lang.NullPointerException

The above test expecting the node list of element `swe:uom`. The new jp2 file "`romagmljp2-collection2-rreq7.jp2`" has the element `swe:uom` but don't have the child elements and the test is trying to iterate on empty NodeList so it throws the null pointer exception.

**Comment by dstenger** [comment]
@keshav-nangare 

**rootInstancecontainsGmlRangeTypeDataRecordUom**
Can you please remove the AssertionError?

**rootInstancecontainsUomByReference**
Can you please add a null check to the test?

**Comment by dstenger** [comment]
@EmDevys 
I just updated the test suite on Beta environment: https://cite.opengeospatial.org/te2/
Now, both files `romagmljp2-collection2-rreq7.jp2` and `roma4326-2b.jp2` are passing all tests. Is this correct from your point of view?

**Comment by keshavnangare** [comment]
> rootInstancecontainsGmlRangeTypeDataRecordUom
> Can you please remove the AssertionError?

> rootInstancecontainsUomByReference
> Can you please add a null check to the test?

@dstenger 

I just took the latest pull as we have fixed #32 and #33. Tested the unit test cases passed successfully with new jp2 file.

Should I update the unit test cases with new jp2 file(roma4326-2b.jp2)?

**Comment by dstenger** [comment]
@keshav-nangare Yes, please do so.

**Comment by keshavnangare** [comment]
Fixed with #34 PR.
